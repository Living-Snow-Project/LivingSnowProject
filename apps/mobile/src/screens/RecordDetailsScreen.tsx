import React from "react";
import { ScrollView, StyleSheet, Text, View } from "react-native";
import {
  AlgaeRecord,
  jsonToRecord,
  recordDateFormat,
} from "@livingsnow/record";
import { RecordDetailsScreenRouteProp } from "../navigation/Routes";
import { Labels } from "../constants/Strings";
import { CachedPhotos } from "../components/CachedPhotos";

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
  },
});

type RecordDetailsScreenProps = {
  route: RecordDetailsScreenRouteProp;
};

export function RecordDetailsScreen({ route }: RecordDetailsScreenProps) {
  const {
    date,
    type,
    name,
    organization,
    locationDescription,
    latitude,
    longitude,
    size,
    colors,
    tubeId,
    notes,
    photos,
  }: AlgaeRecord = jsonToRecord(route.params.record);

  return (
    <ScrollView style={styles.container}>
      <View
        style={{
          borderColor: "black",
          borderWidth: 1,
          borderRadius: 2,
          margin: 2,
        }}
      >
        <View
          style={{
            borderColor: "black",
            borderBottomWidth: 1,
            backgroundColor: "lightblue",
          }}
        >
          <Text style={{ textAlign: "center" }}>Data Sheet</Text>
        </View>
        <Text>{`${Labels.Date}: ${recordDateFormat(date)}`}</Text>
        <Text>{`${Labels.RecordFields.Type}: ${type}`}</Text>
        <Text>{`${Labels.RecordFields.Name}: ${name}`}</Text>
        {!!organization && (
          <Text>{`${Labels.RecordFields.Organization}: ${organization}`}</Text>
        )}
        <Text>{`${Labels.RecordFields.GPSCoordinates}: ${latitude}, ${longitude}`}</Text>
        {!!tubeId && <Text>{`${Labels.RecordFields.TubeId}: ${tubeId}`}</Text>}
        {!!size && <Text>{`${Labels.RecordFields.Size}: ${size}`}</Text>}
        {!!colors && (
          <Text>{`${Labels.RecordFields.Colors}: ${colors.reduce<string>(
            (prev, cur, index) => (index == 0 ? `${cur}` : `${prev}, ${cur}`),
            ""
          )}`}</Text>
        )}
        {!!locationDescription && (
          <Text>{`${Labels.RecordFields.LocationDescription}: ${locationDescription}`}</Text>
        )}
        {!!notes && <Text>{`${Labels.RecordFields.Notes}: ${notes}`}</Text>}
      </View>
      <CachedPhotos photos={photos} />
    </ScrollView>
  );
}
