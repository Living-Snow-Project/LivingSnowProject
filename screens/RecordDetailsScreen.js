import React from 'react';
import { Dimensions, Image, ScrollView, StyleSheet, Text, View } from 'react-native';
import PropTypes from 'prop-types';
import { downloadPhotoUri } from '../lib/Network';
import { getAtlasTypeText } from '../lib/Atlas';

// TODO: make a Record Closure object
function parsePhotoUris(photoUris) {
  if (!photoUris) {
    return [];
  }

  let result = photoUris.split(`;`);
  result.pop();

  return result;
}

function RecordDetailsScreen({route}) {
  const {date, type, name, organization, locationDescription, latitude, longitude, tubeId, notes, atlasType} = route.params.record;
  const photoUris = parsePhotoUris(route.params.record.photoUris);
  // TODO: prefer to scale images based on dominant axis
  const height = Dimensions.get('screen').height * 0.75;
  const atlasTypeText = getAtlasTypeText(atlasType);

  return (
    <ScrollView style={styles.container}>
      <View style={{borderColor: "black", borderWidth: 1, borderRadius: 2, margin: 2}}>
        <View style={{borderColor: "black", borderBottomWidth: 1, backgroundColor: "lightblue"}}>
          <Text style={{textAlign: "center"}}>Data Sheet</Text>
        </View>
        <Text>{`Date: ${date.slice(0, 10)}`}</Text>
        <Text>{`Type: ${type}`}</Text>
        <Text>{`Name: ${name}`}</Text>
        {!!organization && <Text>{`Organization: ${organization}`}</Text>}
        <Text>{`Location: ${latitude}, ${longitude}`}</Text>
        {!!tubeId && <Text>{`TubeId: ${tubeId}`}</Text>}
        {!!atlasType && atlasType > 0 && <Text>{`Atlas Surface Data: ${atlasTypeText}`}</Text>}
        {!!locationDescription && <Text>{`Description: ${locationDescription}`}</Text>}
        {!!notes && <Text>{`Additional Notes: ${notes}`}</Text>}
      </View>
      {photoUris.length > 0 && 
      <View style={{borderColor: "black", borderWidth: 1, borderRadius: 2, margin: 2}}>
        <View style={{borderColor: "black", borderBottomWidth: 1, backgroundColor: "lightblue"}}>
          <Text style={{textAlign: "center"}}>Photos</Text>
        </View>
        <View style={{flex:photoUris.length, flexDirection: "column"}}>
          {photoUris.map((x, index) => 
            <View style={[(index == 0 ? styles.topImage : styles.image), {width: "100%", height: height}]} key={index}>
              <Image style={{width: "100%", height: "100%"}} key={index} source={{uri: downloadPhotoUri(x)}}/>
            </View>)}
        </View>
      </View>}
    </ScrollView>
  );
}

RecordDetailsScreen.propTypes = {
  navigation: PropTypes.object,
  route: PropTypes.object
};

export { RecordDetailsScreen };

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff'
  },
  topImage: {
    flex: 1,
  },
  image: {
    flex: 1,
    borderTopWidth: 1,
    borderColor: "black",
  },
});