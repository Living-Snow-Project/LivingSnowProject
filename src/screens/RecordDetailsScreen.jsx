import React from "react";
import {
  Dimensions,
  Image,
  ScrollView,
  StyleSheet,
  Text,
  View,
} from "react-native";
import PropTypes from "prop-types";
import { downloadPhotoUri } from "../lib/Network";
import { getAtlasPickerItem } from "../record/Atlas";

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
  },
  topImage: {
    flex: 1,
  },
  image: {
    flex: 1,
    borderTopWidth: 1,
    borderColor: "black",
  },
});

// TODO: make an encapsulated Record object
function parsePhotoUris(photoUris) {
  if (!photoUris) {
    return [];
  }

  const result = photoUris.split(`;`);
  result.pop();

  return result;
}

export default function RecordDetailsScreen({ route }) {
  const {
    date,
    type,
    name,
    organization,
    locationDescription,
    latitude,
    longitude,
    tubeId,
    notes,
    atlasType,
  } = route.params.record;
  const photoUris = parsePhotoUris(route.params.record.photoUris);
  // TODO: prefer to scale images based on dominant axis
  const height = Dimensions.get("screen").height * 0.75;

  return (
    <ScrollView style={styles.container}>
      <View
        style={{
          borderColor: "black",
          borderWidth: 1,
          borderRadius: 2,
          margin: 2,
        }}
      >
        <View
          style={{
            borderColor: "black",
            borderBottomWidth: 1,
            backgroundColor: "lightblue",
          }}
        >
          <Text style={{ textAlign: "center" }}>Data Sheet</Text>
        </View>
        <Text>{`Date: ${date.slice(0, 10)}`}</Text>
        {/* BUGBUG: downloaded records type=string, pending records type=enum and converted to string */}
        <Text>{`Type: ${type}`}</Text>
        <Text>{`Name: ${name}`}</Text>
        {!!organization && <Text>{`Organization: ${organization}`}</Text>}
        <Text>{`Location: ${latitude}, ${longitude}`}</Text>
        {!!tubeId && <Text>{`TubeId: ${tubeId}`}</Text>}
        {!!atlasType && atlasType > 0 && (
          <Text>{`Atlas Surface Data: ${
            getAtlasPickerItem(atlasType).label
          }`}</Text>
        )}
        {!!locationDescription && (
          <Text>{`Description: ${locationDescription}`}</Text>
        )}
        {!!notes && <Text>{`Additional Notes: ${notes}`}</Text>}
      </View>
      {photoUris.length > 0 && (
        <View
          style={{
            borderColor: "black",
            borderWidth: 1,
            borderRadius: 2,
            margin: 2,
          }}
        >
          <View
            style={{
              borderColor: "black",
              borderBottomWidth: 1,
              backgroundColor: "lightblue",
            }}
          >
            <Text style={{ textAlign: "center" }}>Photos</Text>
          </View>
          <View style={{ flex: photoUris.length, flexDirection: "column" }}>
            {photoUris.map((x, index) => (
              <View
                style={[
                  index === 0 ? styles.topImage : styles.image,
                  { width: "100%", height },
                ]}
                /* eslint-disable react/no-array-index-key */
                key={index}
              >
                <Image
                  style={{ width: "100%", height: "100%" }}
                  source={{ uri: x.includes(`file`) ? x : downloadPhotoUri(x) }}
                />
              </View>
            ))}
          </View>
        </View>
      )}
    </ScrollView>
  );
}

RecordDetailsScreen.propTypes = {
  // TODO: make an encapsulated Record object (in TypeScript)
  // eslint-disable-next-line react/forbid-prop-types
  route: PropTypes.object.isRequired,
};
