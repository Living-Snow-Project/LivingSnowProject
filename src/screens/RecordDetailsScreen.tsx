import React from "react";
import {
  Dimensions,
  Image,
  ScrollView,
  StyleSheet,
  Text,
  View,
} from "react-native";
import PropTypes from "prop-types";
import { downloadPhotoUri } from "../lib/Network";
import { getAtlasPickerItem } from "../record/Atlas";
import { isAtlas } from "../record/Record";
import { Labels } from "../constants/Strings";

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
  },
  topImage: {
    flex: 1,
  },
  image: {
    flex: 1,
    borderTopWidth: 1,
    borderColor: "black",
  },
});

// TODO: make an encapsulated Record object
function parsePhotoUris(photoUris) {
  if (!photoUris) {
    return [];
  }

  const result = photoUris.split(`;`);
  result.pop();

  return result;
}

export default function RecordDetailsScreen({ route }) {
  const {
    date,
    type,
    name,
    organization,
    locationDescription,
    latitude,
    longitude,
    tubeId,
    notes,
    atlasType,
  } = route.params;
  const photoUris = parsePhotoUris(route.params.photoUris);
  // TODO: prefer to scale images based on its dominant axis
  // which means the server needs to store height, width
  const height = Dimensions.get("screen").height * 0.75;

  return (
    <ScrollView style={styles.container}>
      <View
        style={{
          borderColor: "black",
          borderWidth: 1,
          borderRadius: 2,
          margin: 2,
        }}
      >
        <View
          style={{
            borderColor: "black",
            borderBottomWidth: 1,
            backgroundColor: "lightblue",
          }}
        >
          <Text style={{ textAlign: "center" }}>Data Sheet</Text>
        </View>
        <Text>{`${Labels.RecordFields.Date}: ${date.slice(0, 10)}`}</Text>
        {/* BUGBUG: downloaded records type=string, pending records type=enum and converted to string */}
        <Text>{`${Labels.RecordFields.Type}: ${type}`}</Text>
        <Text>{`${Labels.RecordFields.Name}: ${name}`}</Text>
        {!!organization && (
          <Text>{`${Labels.RecordFields.Organization}: ${organization}`}</Text>
        )}
        <Text>{`${Labels.RecordFields.GPSCoordinates}: ${latitude}, ${longitude}`}</Text>
        {!!tubeId && <Text>{`${Labels.RecordFields.TubeId}: ${tubeId}`}</Text>}
        {isAtlas(type) && (
          <Text>{`${Labels.RecordFields.AtlasSnowSurface}: ${
            // BUGBUG: pickerItem.label... really?
            getAtlasPickerItem(atlasType).label
          }`}</Text>
        )}
        {!!locationDescription && (
          <Text>{`${Labels.RecordFields.LocationDescription}: ${locationDescription}`}</Text>
        )}
        {!!notes && <Text>{`${Labels.RecordFields.Notes}: ${notes}`}</Text>}
      </View>
      {photoUris.length > 0 && (
        <View
          style={{
            borderColor: "black",
            borderWidth: 1,
            borderRadius: 2,
            margin: 2,
          }}
        >
          <View
            style={{
              borderColor: "black",
              borderBottomWidth: 1,
              backgroundColor: "lightblue",
            }}
          >
            <Text style={{ textAlign: "center" }}>
              {Labels.RecordFields.Photos}
            </Text>
          </View>
          <View style={{ flex: photoUris.length, flexDirection: "column" }}>
            {photoUris.map((x, index) => (
              <View
                style={[
                  index === 0 ? styles.topImage : styles.image,
                  { width: "100%", height },
                ]}
                /* eslint-disable react/no-array-index-key */
                key={index}
              >
                <Image
                  style={{ width: "100%", height: "100%" }}
                  source={{ uri: x.includes(`file`) ? x : downloadPhotoUri(x) }}
                />
              </View>
            ))}
          </View>
        </View>
      )}
    </ScrollView>
  );
}

RecordDetailsScreen.propTypes = {
  // TODO: make an encapsulated Record object (in TypeScript)
  // eslint-disable-next-line react/forbid-prop-types
  route: PropTypes.object.isRequired,
};
