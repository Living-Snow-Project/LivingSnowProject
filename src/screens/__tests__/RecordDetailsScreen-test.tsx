import React from "react";
import { render } from "@testing-library/react-native";
import RecordDetailsScreen from "../SettingsScreen";

describe("RecordDetailsScreen test suite", () => {
  test("renders", () => {
    const { toJSON } = render(<RecordDetailsScreen />);

    expect(toJSON()).toMatchSnapshot();
  });
});
