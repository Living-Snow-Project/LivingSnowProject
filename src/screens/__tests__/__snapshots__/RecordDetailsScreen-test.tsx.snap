// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecordDetailsScreen test suite example record with compiled photo 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "borderColor": "black",
          "borderRadius": 2,
          "borderWidth": 1,
          "margin": 2,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "lightblue",
            "borderBottomWidth": 1,
            "borderColor": "black",
          }
        }
      >
        <Text
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Data Sheet
        </Text>
      </View>
      <Text>
        Date: 2022-04-01
      </Text>
      <Text>
        Type: Sample
      </Text>
      <Text>
        Name: Helpful Individual
      </Text>
      <Text>
        Organization: Community Scientist
      </Text>
      <Text>
        Coordinates: 48.06727, -121.12165
      </Text>
      <Text>
        Tube Id: TUBE-1234
      </Text>
      <Text>
        Size: Fist
      </Text>
      <Text>
        Color: Red
      </Text>
      <Text>
        Location Description: White Chuck Glacier, Glacier Peak Wilderness, WA
      </Text>
      <Text>
        Notes: Dark red algae in runnels
      </Text>
    </View>
    <View
      style={
        Object {
          "borderColor": "black",
          "borderRadius": 2,
          "borderWidth": 1,
          "margin": 2,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "lightblue",
            "borderBottomWidth": 1,
            "borderColor": "black",
          }
        }
      >
        <Text
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Photos
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "alignItems": "center",
                "height": 1000.5,
                "justifyContent": "center",
                "width": "100%",
              },
            ]
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`RecordDetailsScreen test suite sample with remote photos 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "borderColor": "black",
          "borderRadius": 2,
          "borderWidth": 1,
          "margin": 2,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "lightblue",
            "borderBottomWidth": 1,
            "borderColor": "black",
          }
        }
      >
        <Text
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Data Sheet
        </Text>
      </View>
      <Text>
        Date: 2021-09-16
      </Text>
      <Text>
        Type: Sample
      </Text>
      <Text>
        Name: test name
      </Text>
      <Text>
        Organization: test org
      </Text>
      <Text>
        Coordinates: -123.456, 96.96
      </Text>
      <Text>
        Tube Id: LAB-1337
      </Text>
      <Text>
        Size: Fist
      </Text>
      <Text>
        Color: Red
      </Text>
      <Text>
        Location Description: test location
      </Text>
      <Text>
        Notes: test notes
      </Text>
    </View>
    <View
      style={
        Object {
          "borderColor": "black",
          "borderRadius": 2,
          "borderWidth": 1,
          "margin": 2,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "lightblue",
            "borderBottomWidth": 1,
            "borderColor": "black",
          }
        }
      >
        <Text
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Photos
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 2,
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "alignItems": "center",
                "height": 1000.5,
                "justifyContent": "center",
                "width": "100%",
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "file:///test-directory/46.jpg",
              }
            }
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "borderColor": "black",
                "borderTopWidth": 1,
                "flex": 1,
              },
              Object {
                "alignItems": "center",
                "height": 1000.5,
                "justifyContent": "center",
                "width": "100%",
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "file:///test-directory/23.jpg",
              }
            }
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`RecordDetailsScreen test suite sample with remote photos, offline mode 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "borderColor": "black",
          "borderRadius": 2,
          "borderWidth": 1,
          "margin": 2,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "lightblue",
            "borderBottomWidth": 1,
            "borderColor": "black",
          }
        }
      >
        <Text
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Data Sheet
        </Text>
      </View>
      <Text>
        Date: 2021-09-16
      </Text>
      <Text>
        Type: Sample
      </Text>
      <Text>
        Name: test name
      </Text>
      <Text>
        Organization: test org
      </Text>
      <Text>
        Coordinates: -123.456, 96.96
      </Text>
      <Text>
        Tube Id: LAB-1337
      </Text>
      <Text>
        Size: Fist
      </Text>
      <Text>
        Color: Red
      </Text>
      <Text>
        Location Description: test location
      </Text>
      <Text>
        Notes: test notes
      </Text>
    </View>
    <View
      style={
        Object {
          "borderColor": "black",
          "borderRadius": 2,
          "borderWidth": 1,
          "margin": 2,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "lightblue",
            "borderBottomWidth": 1,
            "borderColor": "black",
          }
        }
      >
        <Text
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Photos
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 2,
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "alignItems": "center",
                "height": 1000.5,
                "justifyContent": "center",
                "width": "100%",
              },
            ]
          }
        >
          <Text>
            Offline: cannot download photo
          </Text>
          <Text />
        </View>
        <View
          style={
            Array [
              Object {
                "borderColor": "black",
                "borderTopWidth": 1,
                "flex": 1,
              },
              Object {
                "alignItems": "center",
                "height": 1000.5,
                "justifyContent": "center",
                "width": "100%",
              },
            ]
          }
        >
          <Text>
            Offline: cannot download photo
          </Text>
          <Text />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`RecordDetailsScreen test suite sample with remote photos, photo download fails 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "borderColor": "black",
          "borderRadius": 2,
          "borderWidth": 1,
          "margin": 2,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "lightblue",
            "borderBottomWidth": 1,
            "borderColor": "black",
          }
        }
      >
        <Text
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Data Sheet
        </Text>
      </View>
      <Text>
        Date: 2021-09-16
      </Text>
      <Text>
        Type: Sample
      </Text>
      <Text>
        Name: test name
      </Text>
      <Text>
        Organization: test org
      </Text>
      <Text>
        Coordinates: -123.456, 96.96
      </Text>
      <Text>
        Tube Id: LAB-1337
      </Text>
      <Text>
        Size: Fist
      </Text>
      <Text>
        Color: Red
      </Text>
      <Text>
        Location Description: test location
      </Text>
      <Text>
        Notes: test notes
      </Text>
    </View>
    <View
      style={
        Object {
          "borderColor": "black",
          "borderRadius": 2,
          "borderWidth": 1,
          "margin": 2,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "lightblue",
            "borderBottomWidth": 1,
            "borderColor": "black",
          }
        }
      >
        <Text
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Photos
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 2,
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "alignItems": "center",
                "height": 1000.5,
                "justifyContent": "center",
                "width": "100%",
              },
            ]
          }
        >
          <Text>
            There was an error downloading the file: 404
          </Text>
          <Text />
        </View>
        <View
          style={
            Array [
              Object {
                "borderColor": "black",
                "borderTopWidth": 1,
                "flex": 1,
              },
              Object {
                "alignItems": "center",
                "height": 1000.5,
                "justifyContent": "center",
                "width": "100%",
              },
            ]
          }
        >
          <Text>
            There was an error downloading the file: 404
          </Text>
          <Text />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
